engine/userguide/storagedriver/overlayfs-driver.md

```diff
@@ -74,19 +75,7 @@ Before following this procedure, you must first meet all the
     Make sure add this mount to `/etc/fstab` to make it permanent.
 
 4.  Edit `/etc/docker/daemon.json`. If it does not yet exist, create it. Assuming
-    that the file was empty, add the following contents.
-
-    ```json
-    {
-      "storage-driver": "overlay2"
-    }
-    ```
-
-    If you need to use the legacy `overlay` driver, specify it instead.
-
-    To use `overlay2` on CentOS (Docker CE only), you must also set the storage
-    option `overlay2.override_kernel_check`. In this case the `daemon.json`
-    would look like this:
+    that the file was empty, add the following contents for `overlay2`.
```

Comment:
Pretty sure you'd only want the override thing if you are running a kernel prior to 4.0, so it's probably dangerous to recommend users to have it all the time. @cpuguy83 PTAL

Reply:
The kernel-version-check option is no longer needed in current docker versions (at least 17.12, and the upcoming EE release; https://github.com/moby/moby/pull/35527)

Reply:
but it IS required today...

Reply:
With the improved detection, this means that overlay2 is selected by default on;

- Ubuntu, Debian running on kernel 4.x
- CentOS 7
- RHEL 7 (depending on kernel version and backing filesystem)


Running on XFS without `d_type` support is no longer supported, and will now cause docker to _skip_ overlay; existing installs will still run (but still produce an error) to allow users to migrate their data and fix the issue; that will change in a future version, when it will be a fatal error, and docker will refuse to start

Reply:
Great. we can change this doc when it is released...

Reply:
Let's not guide ALL users in overriding the kernel checks when ONLY the users using RHEL on an old kernel need it.

Reply:
@clemenko but it's been released ... in december

Reply:
Correct, these docs are for the latest stable Engine, which is CE 17.12.